src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/components/step_index_pattern/__tests__/step_index_pattern.test.js

```diff
@@ -89,4 +95,26 @@ describe('StepIndexPattern', () => {
     component.setState({ indexPatternExists: true });
     expect(component.find('Header').prop('isNextStepDisabled')).toBe(true);
   });
+
+  it('ensures the response of the latest request is persisted', async () => {
+    const component = createComponent();
+    const instance = component.instance();
+    instance.onQueryChanged({ target: { value: 'e' } });
+    component.setState({ query: 'k' });
+    await new Promise(resolve => process.nextTick(resolve));
+
+    // Honesty, the state would match the result of the `k` query but
+    // it's hard to mock this in tests but if remove our fix
+    // (the early return if the queries do not match) then this
+    // equals [{name: 'es'}]
+    expect(component.state('exactMatchedIndices')).toEqual([]);
+
```

Comment:
I don't think this test is really testing the scenario. Just to be sure, I'd comment out your query staleness checks and confirm this test fails.

Reply:
Yea the assertion doesn't look right but I tried to include a comment indicating why. I couldn't think of a better (or another honestly) way to test this scenario, but if I comment out the early returns, the test definitely fails